controllers/openshift_controller.go

```diff
@@ -416,6 +439,7 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigDeleteRequest() (ctrl.R
 		return ctrl.Result{Requeue: true, RequeueAfter: 15 * time.Second}, nil
 	}
 
+	r.setUpgradeableTrue()
```

Comment:
Any specific reason to not have this statement before the successful return from delete method - https://github.com/openshift/sandboxed-containers-operator/blob/master/controllers/openshift_controller.go#L439 ?

Reply:
in L439 we return because not all machines are updated yet and the uninstall is still in progress. I think this is the right spot to set the upgradeable condition to true.

Reply:
I discussed this with Pradipta offline. The call to r.setUpgradeableTrue() is in this place because it should be done before we update the KataConfig. But that means we should set it to false in case the updates fail. I'll update the code accordingly. Thank you Pradipta!